devdoc/interlocked_linux_requirements.md

```diff
@@ -0,0 +1,369 @@
+```
+**SRS_INTERLOCKED_LINUX_43_064: [** `interlocked_add_64` shall call `atomic_fetch_add` with `addend` as `object` and `value` as `operand`. **]**
+
+**SRS_INTERLOCKED_LINUX_43_065: [** `interlocked_add_64` shall return the initial value of `*addend` plus `value`. **]**
+
+## interlocked_and
+
+```c
+MOCKABLE_FUNCTION(, int32_t, interlocked_and, volatile_atomic int32_t*, destination, int32_t, value);
+```
+**SRS_INTERLOCKED_LINUX_43_003: [** `interlocked_and` shall call `atomic_fetch_and` with `destination` as `object` and `value` as `operand`. **]**
+
+**SRS_INTERLOCKED_LINUX_43_004: [** `interlocked_and` shall return the initial value of `*destination`. **]**
+
+## interlocked_and_16
+
+```c
+MOCKABLE_FUNCTION(, int16_t, interlocked_and_16, volatile_atomic int16_t*, destination, int16_t, value);
+```
+**SRS_INTERLOCKED_LINUX_43_005: [** `interlocked_and_16` shall call `atomic_fetch_and` with `destination` as `object` and `value` as `operand`. **]**
+
+**SRS_INTERLOCKED_LINUX_43_006: [** `interlocked_and_16` shall return the initial value of `*destination`. **]**
+ 
+ ## interlocked_and_64
+
+```c
+MOCKABLE_FUNCTION(, int64_t, interlocked_and_64, volatile_atomic int64_t*, destination, int64_t, value);
+```
+**SRS_INTERLOCKED_LINUX_43_007: [** `interlocked_and_64` shall call `atomic_fetch_and` with `destination` as `object` and `value` as `operand`. **]**
+
+**SRS_INTERLOCKED_LINUX_43_008: [** `interlocked_and_64` shall return the initial value of `*destination`. **]**
+
+## interlocked_and_8
+
+```c
+MOCKABLE_FUNCTION(, int8_t, interlocked_and_8, volatile_atomic int8_t*, destination, int8_t, value);
+```
+**SRS_INTERLOCKED_LINUX_43_009: [** `interlocked_and_8` shall call `atomic_fetch_and` with `destination` as `object` and `value` as `operand`. **]**
+
+**SRS_INTERLOCKED_LINUX_43_010: [** `interlocked_and_8` shall return the initial value of `*destination`. **]**
+
+## interlocked_compare_exchange
+
+```c
+MOCKABLE_FUNCTION(, int32_t, interlocked_compare_exchange, volatile_atomic int32_t*, destination, int32_t, exchange, int32_t, comperand);
+```
+**SRS_INTERLOCKED_LINUX_43_011: [** `interlocked_compare_exchange` shall call `atomic_compare_exchange_strong` with `destination` as `object`, `&comperand` as `expected` and `exchange` as `desired`. **]**
+
+**SRS_INTERLOCKED_LINUX_43_012: [** `interlocked_compare_exchange` shall return a value equal to `comperand` if and only if `atomic_compare_exchange` returns `true`. **]**
+
+**SRS_INTERLOCKED_LINUX_43_066: [** `interlocked_compare_exchange` shall call `atomic_fetch` if `atomic_compare_exchange` returns `false`. **]**
+
+**SRS_INTERLOCKED_LINUX_43_067: [** `interlocked_compare_exchange` shall return the value returned by `atomic_fetch` if the returned value is not equal to `comperand`. **]**
+
+**SRS_INTERLOCKED_LINUX_43_068: [** `interlocked_compare_exchange` shall call `atomic_compare_exchange` if the value returned by `atomic_fetch` is equal to `comperand`. **]**

```

Comment:
>call `atomic_compare_exchange` [](start = 73, length = 30)

I'd say something like "shall repeat the compare exchange if the value returned by `atomic_fetch` is equal to `comperand`"

Because you want to do all of the requirements, not just the compare exchange again.